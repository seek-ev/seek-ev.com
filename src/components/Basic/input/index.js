import React from 'react'

// Styles
import './input.sass'

const Input = (props) => {
  function handleChange(e) {
    props.onChange(e.target)
  }

  return (
    <div className='s-input'>
      <input
        value={props.value}
        onChange={handleChange}
        className={
          (props.raise ? 'raise-input ' : '') +
          (props.border ? 'border-input ' : '') +
          (props.error ? 'error-input ' : '') +
          (props.valid ? 'valid-input ' : '') +
          (props.round ? 'round-input ' : '')
        }
        name={props.name}
        type={props.type}
        placeholder={props.placeholder}
        required={props.required}
        disabled={props.disabled}
        autoComplete={props.autoComplete}
      />

      <div className='error-message'>{props.error}</div>
    </div>
  )
}

export default Input
