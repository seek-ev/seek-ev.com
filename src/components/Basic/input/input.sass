@import "src/assets/variables"

input
    width: 100%
    outline: none
    font-size: 16px
    font-weight: 500
    border-radius: 7px
    text-decoration: none
    box-sizing: border-box
    border: 2px solid #fff
    padding: 7px 8px 7px 7px
    transition: all .25s ease
    font-family: 'Roboto', sans-serif
    box-shadow: -8px 10px 16px 0px rgba(0,0,0,.08)

input:focus
    box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)
    -moz-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)
    -webkit-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)

input:hover
    box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)
    -moz-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)
    -webkit-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)

input:focus:hover::placeholder
    color: #616161

input:hover::placeholder
    color: #080808

input::placeholder
    color: #616161
    font-weight: 500

input:-ms-input-placeholder
    color: #616161
    font-weight: 500

input::-ms-input-placeholder
    color: #616161
    font-weight: 500

input:-webkit-autofill,
input:-webkit-autofill:hover,
input:-webkit-autofill:focus,
input:-webkit-autofill:active
    -webkit-box-shadow: 0 0 0 30px white inset !important

// Disabled
input:disabled
    color: #616161

input:hover:disabled
    color: #616161

input:disabled:hover::placeholder
    color: #616161

input::placeholder:disabled
    color: #616161
    font-weight: 500

input:disabled:hover
    color: #616161
    box-shadow: -8px 10px 16px 0px rgba(0,0,0,.08)

// Seek input
.s-input
    width: 100%
    height: 52px

// Border
.border-input
    border: 2px solid #080808

.border-input:focus
    box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12)
    -moz-box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12)
    -webkit-box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12)

.border-input:hover
    box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12)
    -moz-box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12)
    -webkit-box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12)

// Error input
.error-input
    font-weight: 500
    color: #ed1c46e3
    border: 2px solid rgba(237, 28, 70, 0.9)

.error-input::placeholder
    font-weight: 500
    color: #ed1c46e3

.error-input:-ms-input-placeholder
    font-weight: 500
    color: #ed1c46e3

.error-input::-ms-input-placeholder
    font-weight: 500
    color: #ed1c46e3

.error-input:hover
    font-weight: 500
    color: #ed1c46e3

.error-input:hover::placeholder
    font-weight: 500
    color: #ed1c46e3

.error-message
    max-width: 190px
    font-size: 11px
    overflow: hidden
    color: #ed1c46
    font-weight: bold
    white-space: nowrap
    padding: 0px 0 0 4px
    text-overflow: ellipsis

// Valid input
.valid-input
    border: 2px solid rgba(30, 211, 93, 0.9)

// Animations
.raise-input:focus
    transform: translate(0, -6px)
    box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)
    -moz-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)
    -webkit-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)

.raise-input:hover
    transform: translate(0, -6px)
    box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)
    -moz-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)
    -webkit-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12)

.raise-input:hover::placeholder
    color: #080808

// Round input
.round-input
    padding: 7px 9px
    border-radius: 16px
