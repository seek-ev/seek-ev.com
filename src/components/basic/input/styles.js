import styled from 'styled-components/macro'

const Wrapper = styled.div`
    width: 100%;
    height: 52px;
`

const SInput = styled.input`
  width: 100%;
    outline: none;
    font-size: 16px;
    font-weight: 500;
    border-radius: 7px;
    text-decoration: none;
    box-sizing: border-box;
    border: 2px solid #fff;
    padding: 7px 8px 7px 7px;
    transition: all .25s ease;
    font-family: 'Roboto', sans-serif;
    box-shadow: -8px 10px 16px 0px rgba(0,0,0,.08);
    -moz-box-shadow: -8px 10px 16px 0px rgba(0,0,0,.08);
    -webkit-box-shadow: -8px 10px 16px 0px rgba(0,0,0,.08);

    :focus {
        box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
        -moz-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
        -webkit-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
    }

    :hover {
        box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
        -moz-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
        -webkit-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
    }

    :focus:hover::placeholder {
        color: #616161;
    }

    :hover::placeholder {
        color: #080808;
    }

    ::placeholder {
        color: #616161;
        font-weight: 500;
    }

    :-ms-input-placeholder {
        color: #616161;
        font-weight: 500;
    }

    ::-ms-input-placeholder {
        color: #616161;
        font-weight: 500;
    }

    :-webkit-autofill,
    :-webkit-autofill:hover,
    :-webkit-autofill:focus,
    :-webkit-autofill:active {
        box-shadow: -8px 10px 16px 0px rgba(0,0,0,.08);
        -moz-box-shadow: -8px 10px 16px 0px rgba(0,0,0,.08);
        -webkit-box-shadow: -8px 10px 16px 0px rgba(0,0,0,.08);
    }
    
    :disabled {
        color: #616161;
        opacity: 0.7;
        border: 2px solid #61616100;
    }

    :hover:disabled {
        color: #616161;
    }

    :disabled:hover::placeholder {
        color: #616161;
    }
    
    ::placeholder:disabled {
        color: #616161;
        font-weight: 500;
    }

    :disabled:hover {
        color: #616161;
        box-shadow: -8px 10px 16px 0px rgba(0,0,0,.08);
    }

    ${props => props.raise ? `
        :focus {
            transform: translate(0, -6px);
            box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
            -moz-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
            -webkit-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
        }

        :hover {
            transform: translate(0, -6px);
            box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
            -moz-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
            -webkit-box-shadow: 0px 10px 20px 0px rgba(0,0,0,.12);
        }

        ::placeholder {
            color: #080808;
        }
    ` : ''}

    ${props => props.border ? `
        border: 2px solid #080808;

        :focus {
            box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12);
            -moz-box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12);
            -webkit-box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12);
        }

        :hover {
            box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12)
            -moz-box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12)
            -webkit-box-shadow: 0px 10px 20px 1px rgba(0,0,0,.12)
        }
    ` : ''}

    ${props => props.error ? `
        font-weight: 500;
        color: #ed1c46e3;
        border: 2px solid rgba(237, 28, 70, 0.9);

        ::placeholder {
            font-weight: 500;
            color: #ed1c46e3;
        }

        :-ms-input-placeholder {
            font-weight: 500;
            color: #ed1c46e3;
        }

        ::-ms-input-placeholder {
            font-weight: 500;
            color: #ed1c46e3;
        }

        :hover {
            font-weight: 500;
            color: #ed1c46e3;
        }

        :hover::placeholder {
            font-weight: 500;
            color: #ed1c46e3;
        }
    ` : ''}

    ${props => props.valid ? 'border: 2px solid rgba(30, 211, 93, 0.9)' : ''}

    ${props => props.round ? `
        padding: 7px 9px;
        border-radius: 16px;
        color: red;
    ` : ''}
`

const ErrorMessage = styled.div`
    max-width: 190px;
    font-size: 11px;
    overflow: hidden;
    color: #ed1c46;
    font-weight: bold;
    white-space: nowrap;
    padding: 0px 0 0 4px;
    text-overflow: ellipsis;
`

export { Wrapper, SInput, ErrorMessage }