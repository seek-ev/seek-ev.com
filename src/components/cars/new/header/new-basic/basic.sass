@import 'src/assets/variables'

.new-header-basic
    width: 100%
    display: flex
    flex-direction: column

.new-header-basic-row
    display: flex
    flex-direction: row
    @media only screen and (max-width: $max-desktop)
        flex-direction: column

.new-header-basic-input
    min-width: 260px
    margin: 0 64px 24px 0
    span
        margin: 0 0 3px
        font-weight: 500
    @media only screen and (max-width: $max-medium)
        margin: 0

.new-header-basic-select
    width: 260px
    margin: 0 64px 24px 0
    span
        margin: 0 0 3px
        font-weight: 500
    @media only screen and (max-width: $max-medium)
        width: 100%
        margin: 0 0 12px
