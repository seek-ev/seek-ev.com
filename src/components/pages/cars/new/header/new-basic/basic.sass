@import 'src/assets/variables'

.new-header-basic
    width: 100%
    display: flex
    flex-direction: column

.new-header-basic-row
    display: flex
    flex-direction: row
    justify-content: space-between
    @media only screen and (max-width: $max-medium)
        flex-direction: column

.new-header-basic-input
    min-width: 260px
    margin: 0 64px 24px 0
    span
        margin: 0 0 3px
        font-weight: 500
    @media only screen and (max-width: $max-medium)
        margin: 0

.new-header-basic-input-checkbox
    display: flex
    max-width: 250px
    span
        white-space: nowrap
        margin: 0 8px 3px 0
    input
        -ms-transform: scale(2)
        -moz-transform: scale(2)
        -webkit-transform: scale(2)
        -o-transform: scale(2)
        transform: scale(2)

.new-header-basic-select
    width: 260px
    margin: 0 64px 24px 0
    span
        margin: 0 0 3px
        font-weight: 500
    @media only screen and (max-width: $max-medium)
        width: 100%
        margin: 0 0 12px
