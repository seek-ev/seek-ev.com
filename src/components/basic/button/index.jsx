import React from 'react'

// Styles
import { Wrapper } from './styles'

const Button = (props) => {
  return (
    <Wrapper
      primary={props.primary}
      error={props.error}
      raise={props.raise}
      blue={props.blue}
      dark={props.dark}
      clean={props.clean}
      round={props.round}
      onClick={props.onClick}
      disabled={props.disabled}
    >
      {props.text}
    </Wrapper>
  )
}

export { Button }
