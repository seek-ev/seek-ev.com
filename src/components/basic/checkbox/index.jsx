import React from 'react'

// Styles
import { Wrapper, Title, CheckboxContainer, DefaultCheckbox, StyledCheckbox, Icon } from './styles'

const Checkbox = ({ className, direction, title, checked, name, onChange, disabled }) => {
    const change = (e) => {
        onChange(e.target)
    }

    return (
        <Wrapper className={className} direction={direction}>
            <Title>{title}</Title>
            <CheckboxContainer>
                <DefaultCheckbox checked={checked} name={name} onChange={change} disabled={disabled} />
                <StyledCheckbox checked={checked} disabled={disabled}>
                    <Icon viewBox="0 0 24 24">
                        <polyline points="20 6 9 17 4 12" />
                    </Icon>
                </StyledCheckbox>
            </CheckboxContainer>
        </Wrapper>
    )
}

export { Checkbox }