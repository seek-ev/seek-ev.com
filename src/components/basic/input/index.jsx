import React from 'react'

// Styles
import { Wrapper, Title, SInput, ErrorMessage } from './styles'

const Input = (props) => {
  function handleChange(e) {
    props.onChange(e.target)
  }

  return (
    <Wrapper className={props.className} title={props.title}>
      {props.title ? <Title>
        {props.title}
      </Title> : ''}
      <SInput
        id={props.id}
        value={props.value}
        onChange={handleChange}
        raise={props.raise}
        border={props.border}
        error={props.error}
        valid={props.valid}
        round={props.round}
        name={props.name}
        type={props.type}
        checked={props.checked}
        placeholder={props.placeholder}
        required={props.required}
        disabled={props.disabled}
        autoComplete={props.autoComplete}
        min={props.min}
        max={props.max}
        maxLength={props.maxlength}
        minLength={props.minlength}
        step={props.step}
      />

      <ErrorMessage>{props.error}</ErrorMessage>
    </Wrapper>
  )
}

export { Input }
