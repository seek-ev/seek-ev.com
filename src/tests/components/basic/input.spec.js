import React from 'react'
import { shallow } from 'enzyme'

// Component
import { Input } from 'components/basic/input'

describe('Input Component', () => {
    it('Render input with proper class', () => {
        const input = shallow(<Input />)
        expect(input.find('.seek-input')).toHaveLength(1)
    })

    it('Display placeholder', () => {
        const input = shallow(<Input placeholder="test" />)
        expect(input.find('input').html()).toMatch('placeholder="test"')
    })

    it('onChange event', () => {
        const mockCallback = jest.fn()
        const input = shallow(<Input onChange={mockCallback} />)
        input.find('input').simulate('change', { target: 'value' })
        expect(mockCallback).toBeCalledWith('value')
    })

    it('Set name', () => {
        const input = shallow(<Input name="test" />)
        expect(input.find({ name: 'test' })).toHaveLength(1)
    })

    it('Set type', () => {
        const input = shallow(<Input type="checkbox" />)
        expect(input.find('input').html()).toMatch('type="checkbox"')
    })

    it('Set autocomplete', () => {
        const input = shallow(<Input autoComplete="email" />)
        expect(input.find('input').html()).toMatch('autoComplete="email"')
    })

    it('Value props', () => {
        const input = shallow(<Input value="test" />)
        expect(input.find('input').html()).toMatch('value="test"')
    })

    it('Disabled props', () => {
        const input = shallow(<Input disabled />)
        expect(input.find('input').is('[disabled]')).toBeTruthy()
    })

    it('Raise props', () => {
        const input = shallow(<Input raise />)
        expect(input.find('.raise-input')).toHaveLength(1)
    })

    it('Border props', () => {
        const input = shallow(<Input border />)
        expect(input.find('.border-input')).toHaveLength(1)
    })

    it('Error props', () => {
        const input = shallow(<Input error="Error test" />)
        expect(input.find('.error-input')).toHaveLength(1)
        expect(input.find('.error-message').text()).toEqual('Error test')
    })

    it('Valid props', () => {
        const input = shallow(<Input valid />)
        expect(input.find('.valid-input')).toHaveLength(1)
    })

    it('Round props', () => {
        const input = shallow(<Input round />)
        expect(input.find('.round-input')).toHaveLength(1)
    })

    it('Required props', () => {
        const input = shallow(<Input required />)
        expect(input.find('input').html()).toMatch('required')
    })
})