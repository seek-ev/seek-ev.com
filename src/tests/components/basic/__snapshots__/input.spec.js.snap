// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Render input correctly 1`] = `ShallowWrapper {}`;
